### DISCUSSION TITLE
[BUG] Poco::NamedMutex crash on MacOS 10.15.7

### DISCUSSION THREAD

**Discussion by hdcscy** [pocoproject/poco#4349] [Q&A]
Hello author:
I'm a big fan of the poco library on my work.
Recently, when developing cross-platform applications, I used NamedMutex, and I found that it did not work on macos and crash.

For example:
```c++
// NamedMutex object throws exception when constructing.
Poco::NamedMutex mtx("my name");
if (!mtx.tryLock()) {
     // TODO:
}
```

Poco version: master branch
MacOS version: 10.15.7
CPU architecture: Intel x86_64 & AMD64

**Comment by aleks-f** [comment]
please send a pull request with test case if you want this fixed in the next release

**Comment by matejk** [comment]
Exact code fragment as shown in description does not crash on macOS 14.1 (Apple M1) with clang 15. All named mutex tests pass.

```
Apple clang version 15.0.0 (clang-1500.1.0.2.5)
Target: arm64-apple-darwin23.1.0
Thread model: posix

```

**Reply by hdcscy** [reply]
Hello authors,
My testing situation here:
Still throwing an exception. (the example above)

MacOS version: 10.15.7
CPU architecture: Intel x86_64
Compiler: Apple clang version 12.0

Due to my unfamiliarity with Unix systems, I am unable to locate and fix the issue
Do I need to provide more information?
